src/core/trulens/core/utils/python.py

```diff
@@ -1148,118 +1150,31 @@ def rewrap(obj_):
-    """The class of the singleton instance."""
-
-    frameinfo_codeline: Optional[str]
-    """The frame where the singleton was created.
-
-    This is used for showing "already created" warnings. This is intentionally
-    not the frame itself but a rendering of it to avoid maintaining references
-    to frames and all of the things a frame holds onto.
-    """
-
-    name: Optional[str] = None
-    """The name of the singleton instance.
-
-    This is used for the SingletonPerName mechanism to have a separate singleton
-    for each unique name (and class).
-    """
-
-    def __init__(self, name: str, val: Any):
-        self.val = val
-        self.cls = val.__class__
-        self.name = name
-        self.frameinfo_codeline = code_line(
-            caller_frameinfo(offset=2), show_source=True
-        )
-
-    def warning(self):
-        """Issue warning that this singleton already exists."""
-
-        logger.warning(
-            (
-                "Singleton instance of type %s already created at:\n%s\n"
-                "You can delete the singleton by calling `<instance>.delete_singleton()` or \n"
-                f"""  ```python
-  from trulens.core.utils.python import SingletonPerName
-  SingletonPerName.delete_singleton_by_name(name="{self.name}", cls={self.cls.__name__})
-  ```
-            """
-            ),
-            self.cls.__name__,
-            self.frameinfo_codeline,
-        )
-
-
-class SingletonPerName:
-    """
-    Class for creating singleton instances except there being one instance max,
+    Metaclass for creating singleton instances except there being one instance max,
     there is one max per different `name` argument. If `name` is never given,
     reverts to normal singleton behavior.
     """
 
-    # Hold singleton instances here.
-    _instances: Dict[Hashable, SingletonInfo[SingletonPerName]] = {}
-
-    # Need some way to look up the name of the singleton instance. Cannot attach
-    # a new attribute to instance since some metaclasses don't allow this (like
-    # pydantic). We instead create a map from instance address to name.
-    _id_to_name_map: Dict[int, Optional[str]] = {}
+    _singleton_instances = {}
 
-    def warning(self):
-        """Issue warning that this singleton already exists."""
-
-        name = SingletonPerName._id_to_name_map[id(self)]
-        k = self.__class__.__name__, name
-        if k in SingletonPerName._instances:
-            SingletonPerName._instances[k].warning()
-        else:
-            raise RuntimeError(
-                f"Instance of singleton type/name {k} does not exist."
-            )
-
-    def __new__(
-        cls: Type[SingletonPerName],
-        *args,
-        name: Optional[str] = None,
-        **kwargs,
-    ) -> SingletonPerName:
+    def __call__(cls, *args, name: Optional[str] = None, **kwargs):
         """
         Create the singleton instance if it doesn't already exist and return it.
         """
-
         k = cls.__name__, name
```

Comment:
This was a problem before but might as well fix here. Can you make the first part of the key be "cls.__module__ + "." + cls.__name__` so the same class name in different modules will not cause an issue.